### DISCUSSION TITLE
Catchment delineation D-Inf and/or MDF

### DISCUSSION THREAD

**Discussion by frmgeochile** [wschwanghart/topotoolbox#48] [Questions]
Dear,

   I would like to ask you if it is possible to develop the catchment basin delineation with the D-Infinity and/or MDF method with topotoolbox-matlab. In general I see that it is limited to the D8 method but I ask anyway.

Thank you very much,

Best regards.

**Comment by wschwanghart** [comment]
Hi, catchment delineation is generally limited to single flow directions if you want that any location can be assigned to only one outlet location. This is not true if you have multiple flow directions that will eventually link one pixel to many outlets, for example if these pixels are located on a divide. Some years back, I have developed an approach that interprets the fractions in multiple flow directions as probabilities. Based on these probabilities, one can calculate the probability of a pixel to drain towards a specific outlet. This can be repeated for all outlets. Based on these probabilities, one can then obtain a metric how well a pixel can be assigned to an outlet, the **drainage basin entropy**. 

This approach is implemented in the function `FLOWobj/dbentropy` and the accompanying paper [can be found here](http://dx.doi.org/10.1016/j.envsoft.2012.01.016).

Reference:
Schwanghart, W., Heckmann, T. (2012): Fuzzy delineation of drainage basins through probabilistic interpretation of diverging flow algorithms. Environmental Modelling & Software, 33, 106-113. [DOI: [10.1016/j.envsoft.2012.01.016](http://dx.doi.org/10.1016/j.envsoft.2012.01.016)]
